test/rulesets/Base/base.jl

```diff
@@ -147,22 +147,13 @@
     end
 
     @testset "hypot(x, y)" begin
-        x, y = rand(2)
-        h, pushforward = frule(hypot, x, y)
-        dxy(x, y) = pushforward(NamedTuple(), x, y)
-
-        @test extern(dxy(One(), Zero())) === x / h
-        @test extern(dxy(Zero(), One())) === y / h
-
-        cx, cy = cast((One(), Zero())), cast((Zero(), One()))
-        dx, dy = extern(dxy(cx, cy))
-        @test dx === x / h
-        @test dy === y / h
-
-        cx, cy = cast((rand(), Zero())), cast((Zero(), rand()))
-        dx, dy = extern(dxy(cx, cy))
-        @test dx === x / h * cx.value[1]
-        @test dy === y / h * cy.value[2]
+        rng = MersenneTwister(123456)
+        x, Δx, x̄ = randn(rng, 3)
+        y, Δy, ȳ = randn(rng, 3)
+        Δz = randn(rng)
+
+        frule_test(hypot, (x, Δx), (y, Δy))
+        rrule_test(hypot, Δz, (x, x̄), (y, ȳ))
```

Comment:
if there a reason these use `frule_test` but the tests for e.g. `atan2` call `frule` then test the output.

Is it because the `atan2` tests pre-date `frule_test`?

Reply:
> Is it because the atan2 tests pre-date frule_test?

I suspect that this is the case. Will open an issue and resolve at a later date.